api/src/main/java/net/md_5/bungee/api/connection/PendingConnection.java

```diff
@@ -89,4 +90,15 @@ public interface PendingConnection extends Connection
      * @return Whether the client is using a legacy client.
      */
     boolean isLegacy();
+
+    /**
+     * Retrieves a cookie from this pending connection.
+     *
+     * @param cookie the resource location of the cookie, for example "bungeecord:my_cookie"
+     *
+     * @return a {@link CompletableFuture} that will be completed when the Cookie response is received
+     * Only useable for 1.20.5 clients or newer during the login, config or game phase
+     * if the cookie is not set in the client, the {@link CompletableFuture} will complete with a null value
+     */
+    CompletableFuture<byte[]> retrieveCookie(String cookie);
```

Comment:
Nice use of CompletableFuture!

Can you please add `@ApiStatus.Experimental` annotation?

You'll need to merge in master as I just added the dependency in c69acf728c75b98070c225595bccc682ded13b74

Reply:
done